kernel/src/vm/vmar/vm_mapping.rs

```diff
@@ -163,6 +164,35 @@ impl VmMapping {
             MappedMemory::Vmo(_) | MappedMemory::Device => RssType::RSS_FILEPAGES,
         }
     }
+
+    /// Populates device memory for this mapping.
+    ///
+    /// This method should only be called for device memory mappings. It maps
+    /// the provided I/O memory region into the virtual address space.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if the mapping is not a device memory mapping.
+    pub(super) fn populate_device(
+        &self,
+        vm_space: &VmSpace,
+        io_mem: IoMem,
+        vmo_offset: usize,
+    ) -> Result<()> {
+        debug_assert!(matches!(self.mapped_mem, MappedMemory::Device));
+        if !matches!(self.mapped_mem, MappedMemory::Device) {
+            panic!("populate_device called on non-device mapping");
```

Comment:
No need to do this. This is an internal method so I think `debug_assert!` is enough.